resourcemanager/pooltask/task_manager_iterator.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pooltask
+
+import (
+	"container/list"
+	"time"
+)
+
+func (t *TaskManager[T, U, C, CT, TF]) getBoostTask() (tid uint64, result *TaskBox[T, U, C, CT, TF]) {
+	// boost task
+	// 1、the count of running task is less than concurrency
+	// 2、less run time, more possible to boost
+	tid, element := t.iter(canBoost[T, U, C, CT, TF])
+	if element != nil {
+		return tid, element.Value.(tContainer[T, U, C, CT, TF]).task
+	}
+	return 0, nil
+}
+
+func (t *TaskManager[T, U, C, CT, TF]) pauseTask() {
+	// pause task,
+	// 1、more run time, more possible to pause
+	// 2、if task have been boosted, first to pause.
+	tid, result := t.iter(canPause[T, U, C, CT, TF])
+	if result != nil {
+		result.Value.(tContainer[T, U, C, CT, TF]).task.status.CompareAndSwap(RunningTask, StopTask)
+		// delete it from list
+		shardID := getShardID(tid)
+		t.task[shardID].rw.Lock()
+		defer t.task[shardID].rw.Unlock()
+		t.task[shardID].stats[tid].stats.Remove(result)
+	}
+}
+
+func (t *TaskManager[T, U, C, CT, TF]) iter(fn func(m *meta[T, U, C, CT, TF], max time.Time) (*list.Element, bool)) (tid uint64, result *list.Element) {
+	var compareTS time.Time
+	for i := 0; i < shard; i++ {
+		breakFind := func(index int) (breakFind bool) {
+			t.task[i].rw.RLock()
+			defer t.task[i].rw.RUnlock()
+			for id, stats := range t.task[i].stats {
+				if result == nil {
+					result = findTask[T, U, C, CT, TF](stats, RunningTask)
+					tid = id
+					compareTS = stats.createTS
+					continue
+				}
+				newResult, pauseFind := fn(stats, compareTS)
+				if pauseFind {
+					result = newResult
```

Comment:
If here we call `canBoost`, and we return `nil, true` https://github.com/pingcap/tidb/pull/40670/files#diff-f1e67ff3fe9cec68e358a2e0ab7bc91ecca0dfcee2e9827282a0b6c060449a64R101. Then `newResult` is nil, is this expected and why？

Reply:
Fixed, it is a misjudgment.